lib/teleterm/daemon/daemon.go

```diff
@@ -139,13 +139,23 @@ func (s *Service) CreateGateway(ctx context.Context, params clusters.CreateGatew
 }
 
 // createGateway assumes that mu is already held by a public method.
-func (s *Service) createGateway(ctx context.Context, params clusters.CreateGatewayParams) (*gateway.Gateway, error) {
+func (s *Service) createGateway(ctx context.Context, params CreateGatewayParams) (*gateway.Gateway, error) {
 	cluster, err := s.ResolveCluster(params.TargetURI)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	gateway, err := cluster.CreateGateway(ctx, params)
+	cliCommandProvider := clusters.NewDbcmdCLICommandProvider(s.Storage)
```

Comment:
> * How will this be handled in the future when `cliCommandProvider` should be a `clusters.NewAppcmdCLICommandProvider`?

We know what resource type the gateway targets through `params.TargetURI`. In case of a db it's something like `/clusters/teleport-local/dbs/postgres`. We could inspect that and pass an appropriate `cliCommandProvider`.

> * Does it make sense for every gateway to use this db command provider? If this gateway isn't doing any thing with `db` should this be a noop?

For now Connect supports only db gateways, so I hardcoded this single command provider here.

---

In general, I plan to add a readme to lib/teleterm which would explain the general architecture and a couple of the concepts used in Connect. With the release coming up soon there's been always something seemingly more important to do, but with new people joining the team it makes even more sense to do that. This should help other teammates during code reviews as well, as other than Grzegorz and I people don't usually work on lib/teleterm.